### ISSUE TITLE
Connection Issue - Peer disconnected during the middle of a handshake (iOS) 

### ISSUE THREAD

**Issue by adarshm** [clostra/newnode#96]
We are facing an issue while using newnode in one of our iOS projects. We are getting an error response from the API call. Upon investigation, we found the following in the logs

**"Peer disconnected during the middle of a handshake"**

Use case:

1. Open the application and make an api call - **Works**
2. Hit the home button and **lock the device**
3. Unlock the device and open the application again - **API failure**

This issue can be reproduced with an **iOS device** through the above steps.

We have created a sample application to demo this and is available in the following repo: 

https://github.com/adarshm/newnode-test-pro  (updated with SPM)

Here are the error logs:

```
2020-11-02 16:57:27.137682+0530 NewnodeTest[711:97340] [] nw_socket_handle_socket_event [C8.1:2] Socket SO_ERROR [61: Connection refused]
2020-11-02 16:57:27.148168+0530 NewnodeTest[711:97340] [] nw_socket_handle_socket_event [C8.2:4] Socket SO_ERROR [61: Connection refused]
2020-11-02 16:57:27.148704+0530 NewnodeTest[711:97340] [BoringSSL] nw_protocol_boringssl_error(1584) [0x1011261b0] Lower protocol stack error: 61
2020-11-02 16:57:27.149524+0530 NewnodeTest[711:97340] [BoringSSL] nw_protocol_boringssl_input_finished(1543) [0x1011261b0] Peer disconnected during the middle of a handshake. Sending errSSLClosedNoNotify(-9816) alert
2020-11-02 16:57:27.153228+0530 NewnodeTest[711:97340] TIC TCP Conn Failed [8:0x2835fa7c0]: 1:61 Err(61)
2020-11-02 16:57:27.159419+0530 NewnodeTest[711:97340] Tube connection 0x101015be8 received error 1:61 => proxy error 4:-2096
2020-11-02 16:57:27.164318+0530 NewnodeTest[711:97340] [] nw_socket_handle_socket_event [C9.1:2] Socket SO_ERROR [61: Connection refused]
2020-11-02 16:57:27.168309+0530 NewnodeTest[711:97340] [] nw_socket_handle_socket_event [C9.2:4] Socket SO_ERROR [61: Connection refused]
2020-11-02 16:57:27.168415+0530 NewnodeTest[711:97340] [BoringSSL] nw_protocol_boringssl_error(1584) [0x101130dd0] Lower protocol stack error: 61
2020-11-02 16:57:27.168863+0530 NewnodeTest[711:97340] [BoringSSL] nw_protocol_boringssl_input_finished(1543) [0x101130dd0] Peer disconnected during the middle of a handshake. Sending errSSLClosedNoNotify(-9816) alert
2020-11-02 16:57:27.171319+0530 NewnodeTest[711:97340] TIC TCP Conn Failed [9:0x2835f43c0]: 1:61 Err(61)
2020-11-02 16:57:27.175157+0530 NewnodeTest[711:97340] Task <30B38F74-5CFA-4699-8704-1CBFF53D315B>.<1> HTTP load failed (error code: -1004 [1:61])
2020-11-02 16:57:27.176273+0530 NewnodeTest[711:96032] Task <30B38F74-5CFA-4699-8704-1CBFF53D315B>.<1> finished with error - code: -1004
Error Error Domain=NSURLErrorDomain Code=-1004 "Could not connect to the server." UserInfo={NSUnderlyingError=0x280ea2100 {Error Domain=kCFErrorDomainCFNetwork Code=-1004 "(null)" UserInfo={_kCFStreamErrorCodeKey=61, _kCFStreamErrorDomainKey=1}}, NSErrorFailingURLStringKey=xxxxxxxxAPI_URLxxxxxxxxx, NSErrorFailingURLKey=xxxxxxxxAPI_URLxxxxxxxxx, _kCFStreamErrorDomainKey=1, _kCFStreamErrorCodeKey=61, NSLocalizedDescription=Could not connect to the server.}

```
Please let us know if we have to do any additional configurations to make this work. 

Note: We have integrated the NewNode lib as a Swift Package (using SPM).

**Comment by adarshm** [comment]
Hi NewNode Team, 

Please don't close this issue, this issue is still occurring with the latest version of NewNode iOS SDK. 

Have you got a chance to review this? Please give more priority to this. I’m eager to receive your response.

Thanks in advance.

**Comment by adarshm** [comment]
Please note that this issue is still there and we are waiting for a solution from you. Thanks in advance!

**Comment by Korzhuck** [comment]
I am faced this issue too.

**Comment by ghazel** [comment]
Fixed. Will be released shortly.

**Comment by ghazel** [comment]
Released in [2.0.4](https://github.com/clostra/newnode/releases/tag/2.0.4)
